beta-src/src/utils/map/drawArrow.ts

```diff
@@ -17,7 +17,106 @@ export default function drawArrow(
 ): void {
   const d3MapSelector = d3.select("#map");
 
+  let x1;
+  let x2;
+  let y1;
+  let y2;
+
   switch (receiverType) {
+    case "unit": {
+      const toTerritoryName = Territory[receiverIdentifier];
+      const fromTerritoryName = Territory[unitTerritory];
+      let unitSlotEl = d3.select(`#${fromTerritoryName}-unit`).node();
+      let receiverUnit = d3.select(`#${toTerritoryName}-unit`).node();
+      const fromTerritoryEl: SVGSVGElement = d3
+        .select(`#${fromTerritoryName}-territory`)
+        .node();
+      const toTerritoryEl: SVGSVGElement = d3
+        .select(`#${toTerritoryName}-territory`)
+        .node();
+
+      if (fromTerritoryEl && toTerritoryEl && receiverUnit && unitSlotEl) {
+        unitSlotEl = unitSlotEl.parentNode;
+        receiverUnit = receiverUnit.parentNode;
+        const unitSlotElBBox = unitSlotEl.getBBox();
```

Comment:
can we just desctructure here  instead`const {elx, ely, elwidth, elheight } = unitSlotEl.getBBox();`